install.sh

```diff
@@ -59,27 +59,31 @@ case $PATH in
         ;;
 esac
 
-printf "Install location [default: %s]: " "$default_bin"
-read -r bindir < /dev/tty
-bindir=${bindir:-$default_bin}
-
-while ! test -d "$bindir"; do
-    echo "Directory $bindir does not exist"
+_read_bindir() {
     printf "Install location [default: %s]: " "$default_bin"
-    read -r bindir < /dev/tty
-    bindir=${bindir:-$default_bin}
-done
+    read -r XH_BINDIR < /dev/tty
+    XH_BINDIR=${XH_BINDIR:-$default_bin}
+}
+
+if [ -z "$XH_BINDIR" ]; then
+    _read_bindir
+
+    while ! test -d "$XH_BINDIR"; do
+      echo "Directory $XH_BINDIR does not exist"
+      _read_bindir
+    done
+fi
 
 tar xzf xh.tar.gz
 
-if test -w "$bindir"; then
-    mv xh-*/xh "$bindir/"
-    ln -sf "$bindir/xh" "$bindir/xhs"
+if test -w "$XH_BINDIR"; then
+    mv xh-*/xh "$XH_BINDIR/"
+    ln -sf "$XH_BINDIR/xh" "$XH_BINDIR/xhs"
 else
-    sudo mv xh-*/xh "$bindir/"
-    sudo ln -sf "$bindir/xh" "$bindir/xhs"
+    sudo mv xh-*/xh "$XH_BINDIR/"
+    sudo ln -sf "$XH_BINDIR/xh" "$XH_BINDIR/xhs"
```

Comment:
If `$XH_BINDIR` does not exist we take this branch, which doesn't help.

Maybe we wouldn't want to use sudo at all if the environment variable was used, to guarantee it's non-interactive.

Perhaps:
- Go back to using `$bindir`, with an explicit `bindir=$XH_BINDIR`
- Change the condition to `test -w "$bindir" || [ -n "$XH_BINDIR" ]`, so it doesn't try to be smart